velero-2-mixin/variables.libsonnet

```diff
@@ -0,0 +1,95 @@
+      local groupLabels = this.config.groupLabels,
+      local instanceLabels = this.config.instanceLabels,
+      local topClusterSelector =
+        var.custom.new(
+          'top_cluster_count',
+          values=[2, 3, 6, 8, 10],
+        )
+        + var.custom.generalOptions.withDescription(
+          'This variable allows for modification of top cluster value.'
+        )
+        + var.custom.generalOptions.withLabel('Top cluster count'),
+      local root = self,
+      local variablesFromLabels(groupLabels, instanceLabels, filteringSelector, multiInstance=true) =
+        local chainVarProto(index, chainVar) =
+          var.query.new(chainVar.label)
+          + var.query.withDatasourceFromVariable(root.datasources.prometheus)
+          + var.query.queryTypes.withLabelValues(
+            chainVar.label,
+            '%s{%s}' % [varMetric, chainVar.chainSelector],
+          )
+          + var.query.generalOptions.withLabel(utils.toSentenceCase(chainVar.label))
+          + var.query.selectionOptions.withIncludeAll(
+            value=if (!multiInstance && std.member(instanceLabels, chainVar.label)) then false else true,
+            customAllValue='.+'
+          )
+          + var.query.selectionOptions.withMulti(
+            if (!multiInstance && std.member(instanceLabels, chainVar.label)) then false else true,
+          )
+          + var.query.refresh.onTime()
+          + var.query.withSort(
+            i=1,
+            type='alphabetical',
+            asc=true,
+            caseInsensitive=false
+          );
+        std.mapWithIndex(chainVarProto, utils.chainLabels(groupLabels + instanceLabels, [filteringSelector])),
+      datasources: {
+        prometheus:
+          var.datasource.new('prometheus_datasource', 'prometheus')
+          + var.datasource.generalOptions.withLabel('Prometheus data source')
+          + var.datasource.withRegex(''),
+        loki:
+          var.datasource.new('loki_datasource', 'loki')
+          + var.datasource.generalOptions.withLabel('Loki data source')
+          + var.datasource.withRegex('')
+          // hide by default (used for annotations)
+          + var.datasource.generalOptions.showOnDashboard.withNothing(),
+      },
+      // Use on dashboards where multiple entities can be selected, like fleet dashboards
+      multiInstance:
+        [root.datasources.prometheus]
+        + variablesFromLabels(groupLabels, instanceLabels, filteringSelector),
+      multiCluster:
+        [root.datasources.prometheus]
+        + variablesFromLabels(groupLabels, [], filteringSelector),
+      // Use on dashboards where only single entity can be selected, like drill-down dashboards
+      singleInstance:
+        [root.datasources.prometheus]
+        + variablesFromLabels(groupLabels, instanceLabels, filteringSelector, multiInstance=false),
+
+      queriesSelector:
+        '%s,%s' % [
+          utils.labelsToPromQLSelector(groupLabels + instanceLabels),
+          filteringSelector,
```

Comment:
I assumed this is how we had to do it based off the examples in helloworld-observ-lib and windows-observ-lib

Reply:
I'll change it here for now and send a message to Vitaly to check why he added that second. It seems like a really specific thing to add across the different mixins so i can't imagine it would be a mistake, but maybe.